Make library docs more FAQ-like (#1164)
python/typing: docs/source/libraries.rst

distributed package. If using ``setuptools``, this can be achieved like so:

.. code-block:: python

   from setuptools import setup

   setup(
      name="my_great_distribution",
      version="0.1",
      package_data={"my_great_package": ["py.typed"]},
      packages=["my_great_package"],
   )

Type stub files included in the package
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

It's possible to include a mix of type stub files (``.pyi``) and inline type
annotations (``.py``). One use case for including type stub files in your
package is to provide types for extension modules in your library. A typical
directory structure would look like:

.. code-block:: text

   setup.py
   my_great_package/
      __init__.py
      stuff.py
      stuff.pyi
      py.typed

If using ``setuptools``, we can ensure the ``.pyi`` and ``py.typed`` files are
included like so:

.. code-block:: python

   from setuptools import setup

   setup(
      name="my_great_distribution",
      version="0.1",
      package_data={"my_great_package": ["py.typed", "stuff.pyi"]},
      packages=["my_great_package"],
   )

The presence of ``.pyi`` files does not affect the Python interpreter at runtime

> in any way. However, static type checkers will only look at ``.pyi`` file and

Reviewer: ```suggestion
in any way. However, static type checkers will only look at the ``.pyi`` file and
```

ignore the corresponding ``.py`` file.

Companion type stub package
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> These are often referred to as "stub-only" packages. The package should have a
> prefix of the runtime package name with a suffix of ``-stubs``. The py.typed

Reviewer: Not a fan of this sentence:
```suggestion
These are often referred to as "stub-only" packages. The name of the stub package
should be the name of the runtime package suffixed with ``-stubs``. The ``py.typed``
```

marker file is not necessary for stub-only packages. This approach can be useful
to develop type stubs independently from your library.

For example:

.. code-block:: text

   setup.py
   my_great_package-stubs/
      __init__.pyi
      stuff.pyi

.. code-block:: python

   from setuptools import setup

   setup(
      name="my_great_package-stubs",
      version="0.1",
      package_data={"my_great_package-stubs": ["__init__.pyi", "stuff.pyi"]},
      packages=["my_great_package-stubs"]
   )

Users are then able to install the stubs-only package separately to provide
types for the original library.

> Note that to ensure inclusion of ``.pyi`` and ``py.typed`` files in an sdist

Reviewer: Weird that this is in the stub package section, while the one about inline stub files only talks about `setup.py`.
